File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/RunTarget.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum RunTarget {
    Cloud,
    Runner;

    public String toWire() {
        return switch (this) {
            case Cloud -> "cloud";
            case Runner -> "runner";
        };
    }

    public static RunTarget fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "cloud" -> Cloud;
            case "runner" -> Runner;
            default -> throw new IllegalArgumentException("Unknown run target: " + value);
        };
    }
}

File: apps/backend/src/Common/Domain/ValueObjects/RunTarget.cs
namespace OffceOs.Domain.Common.ValueObjects;

public enum RunTarget
{
    Cloud,
    Runner,
}

public static class RunTargetExtensions
{
    public static string ToStorageString(this RunTarget target) => target switch
    {
        RunTarget.Cloud => "cloud",
        RunTarget.Runner => "runner",
        _ => throw new ArgumentOutOfRangeException(nameof(target)),
    };

}
